File: src/main/kotlin/io/shaded/config/impl/RegionConfig.java
package io.shaded.config.impl;

import io.shaded.ShadedRegionPlugin;

import java.util.Arrays;
import java.util.List;

public class RegionConfig {

    private static final transient RegionConfig instance = new RegionConfig();

    public static List<String> regions = Arrays.asList("spawn");

    public static void load() {
        ShadedRegionPlugin.getConfigSerializer().load(instance, RegionConfig.class, "region-config");
    }

    public static void save() {
        ShadedRegionPlugin.getConfigSerializer().save(instance, "region-config");
    }

}

File: src/main/kotlin/io/shaded/config/impl/RegionConfig.kt
package io.shaded.config.impl

import io.shaded.ShadedRegionPlugin.Companion.configSerializer
import java.util.*

class RegionConfig : Config() {

    fun load() {
        configSerializer!!.load(instance, Config::class.java, "region-config")
    }

    fun save() {
        configSerializer!!.save(instance, "region-config")
    }

    companion object {
        @Transient
        val instance = RegionConfig()
    }

}


open class Config {
    var regions = Arrays.asList("spawn")
}

